### PR TITLE
[#30] Complete the script to generate the whole KMM project

### LINKED ISSUE #30: Build the script to generate the whole KMM project
## Why

After finalizing the KMM Templates's project structure, we should build a script to help the developers to generate the new KMM project with some initial input parameters, such as project name, app name, app id, etc.

- As leveraging Kotlin is the main idea of KMM, we should try to use Kotlin as much as possible as the main script's language.
- We should find a way to make the template to be **configurable** such as using [mustache](https://mustache.github.io/) template using manually using text replacement with a high risk of mistakes of wrong content generating.
- The script will generate the whole KMM project from `android`, and `ios` to the `shared` module.

## Who Benefits?

Developers

### PR DESCRIPTION
**Pull request by luongvo**
- Close #30

## What happened 👀

- Switch the android submodule to the [develop-koin](https://github.com/nimblehq/android-templates/tree/develop-koin) branch.
- Fix: generate the Android app module with the entered bundle id, not always to `co.nimblehq.kmm.template`.
- Create a script to generate the `shared` module.
- Move `make_android` and `make_ios` scripts to the scripts folder.
- Customize the output folder to the `project_name`.
- Update README instructions.

## Insight 📝

- To generate the `shared` module with the coresponding package name & project name, clone & adjust the [Android Templates's kscript](https://github.com/nimblehq/android-templates/blob/develop-koin/scripts/new_project.kts) to generate only the `shared` module.
- To customize the output folder, replace all `sample` folder name with `$project_name`.

## Proof Of Work 📹

> I temporarily switched to generate the sample project with a custom package name as `co.nimblehq.kmm.sample` to demonstrate that the script can now generate the whole KMM project with entered inputs ✅ 

- Please verify through this commit https://github.com/nimblehq/kmm-templates/pull/47/commits/16a34a6014c082252e3f7ceb7a6f25c9e0e3b4ff.

> We will switch it back to `co.nimblehq.kmm.template` after having enough approvals.

- The script now is able to generated the new project to a custom folder as `project_name`, not always to the `sample` folder 🎉  For example, running `./make.sh --bundle-id co.nimblehq.kmm.sample --bundle-id-staging co.nimblehq.kmm.sample.staging --project-name sample2 --ios-version 14.0` will generate a new project at `sample2` folder.

Pasted_Image_02_11_2023__00_42

### CODE REVIEW

File: sample/android/src/main/java/co/nimblehq/kmm/sample/lib/TypeAlias.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package co.nimblehq.kmm.sample.lib
+
+typealias IsLoading = Boolean
```
Comment (markgravity) [inline sample/android/src/main/java/co/nimblehq/kmm/sample/lib/TypeAlias.kt:3]: This naming is weird, where is the use case of it? 🤔

**Review by markgravity — COMMENTED** [review]
The rest LGTM 🙏

**Review by lydiasama — APPROVED** [review]

**Comment by luongvo** [comment]
TODO: missing `buildKonfig.properties` https://github.com/nimblehq/kmm-templates/tree/feature/30-complete-the-script-to-generate-new-kmm-project/sample

**Comment by luongvo** [comment]
> TODO: missing `buildKonfig.properties` [`feature`/30-complete-the-script-to-generate-new-kmm-project/sample](https://github.com/nimblehq/kmm-templates/tree/feature/30-complete-the-script-to-generate-new-kmm-project/sample?rgh-link-date=2023-11-07T04%3A49%3A48Z)

Fixed in https://github.com/nimblehq/kmm-templates/pull/47/commits/f1191bfe814801908d134353e0c2d7420859b217

File: sample/android/src/main/java/co/nimblehq/kmm/sample/lib/TypeAlias.kt (hunk 1 above)
Comment (luongvo) [reply sample/android/src/main/java/co/nimblehq/kmm/sample/lib/TypeAlias.kt:3]: @markgravity It's an alias of the loading state. We're using it in Android Templates and applied to KMM Templates too https://github.com/nimblehq/kmm-templates/blob/9bf253c53a000a99f6ef59b15251a7fb532d4af7/sample/android/src/main/java/co/nimblehq/kmm/template/ui/base/BaseViewModel.kt#L17-L18

**Review by markgravity — APPROVED** [review]

**Pull request merged**